lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -6661,6 +6661,48 @@ class DecomposeAtenRot90Op : public OpRewritePattern<AtenRot90Op> {
 };
 } // namespace
 
+// Decompose aten.count_nonzero to aten.ne.Scalar and aten.sum.dim_IntList
+namespace {
+class DecomposeAtenCountNonzeroDimIntListOp
+    : public OpRewritePattern<AtenCountNonzeroDimIntListOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+  LogicalResult matchAndRewrite(AtenCountNonzeroDimIntListOp op,
+                                PatternRewriter &rewriter) const override {
+    auto dimList = op.getDim();
+    if (!isa<Torch::ListType>(dimList.getType()) &&
+        !isa<Torch::IntType>(
+            (cast<Torch::ListType>(dimList.getType())).getContainedType())) {
+      return rewriter.notifyMatchFailure(
+          op, "expected `dim` to be constructed from list");
+    }
+    SmallVector<Value> dimListElements;
+    if (!getListConstructElements(dimList, dimListElements)) {
+      return rewriter.notifyMatchFailure(
+          op, "expected `dim` to be constructed from list");
+    }
```

Comment:
Instead of doing all this, you can just add a check for none and if found non-none then just use: `m_TorchListOfConstantInts`.